### PR TITLE
Migration to new block-service and block interface

### PR DESCRIPTION
**Pull request by dignifiedquire**
Ref https://github.com/ipfs/js-ipfs/issues/787

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -25,11 +25,18 @@ function noop () {}
...
     }
+    // sorry nicola need to async init and open the repo
+    // nicola will love this!
+    // if (!blockService) {
+    //   const repo = new IPFSRepo('in-memory', {
+    //     fs: MemoryStore,
+    //     level: require('memdown')
+    //   })
+    //   blockService = new BlockService(repo)
+    // }
```
Comment (diasdavid) [inline src/index.js]: @nicola will not like that

Comment (diasdavid) [inline src/index.js]: but yeah, understood. Do what you gotta do.

**Comment by dignifiedquire** [comment]
@diasdavid the only failing test is because of https://github.com/dignifiedquire/js-ipld-eth-state-trie/tree/datastore 

There is a new ipfs block created in the resolver and I'm unsure what the right CID to put in there is. Could you take a look please?

File: src/index.js [hunk 2]
```diff
@@ -366,32 +328,50 @@ class IPLDResolver {
...
+  series([
+    (cb) => repo.init({}, cb),
+    (cb) => repo.open(cb)
+  ], (err) => {
+    if (err) {
+      return callback(err)
+    }
+    callback(null, new IPLDResolver(blockService))
+  })
+}
```
Comment (diasdavid) [inline src/index.js]: so that @nicola remains happy 🙌🏽

File: src/index.js [hunk 3]
```diff
@@ -25,10 +26,8 @@ function noop () {}
 
 class IPLDResolver {
   constructor (blockService) {
-    // nicola will love this!
```
Comment (diasdavid) [inline src/index.js]: poor @nicola

File: test/example-repo/datastore/LOG.old [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-07:49:57.064655 db@close done T·72.59µs
+07:50:02.056578 log@legend F·NumFile S·FileSize N·Entry C·BadEntry B·BadBlock Ke·KeyError D·DroppedEntry L·Level Q·SeqNum T·TimeElapsed
+07:50:02.057231 db@open opening
+07:50:02.057312 journal@recovery F·1
+07:50:02.057514 journal@recovery recovering @3
+07:50:02.058921 mem@flush created L0@5 N·4 S·1KiB "/ip..\xf6\xe4\xa9,v5":"/pk..\xf6\xe4\xa9,v6"
+07:50:02.059983 db@janitor F·4 G·0
+07:50:02.060001 db@open done T·2.755926ms
+07:50:02.073183 db@close closing
+07:50:02.073285 db@close done T·97.522µs
```
Comment (diasdavid) [inline test/example-repo/datastore/LOG.old]: add to gitignore

File: src/index.js (hunk 2 above)
Comment (dignifiedquire) [inline src/index.js]: this turned out to be great, as I had forgotten to test in-memory in nodejs and found a bug in the repo

**Comment by diasdavid** [comment]
@dignifiedquire this one? https://github.com/dignifiedquire/js-ipld-eth-state-trie/blob/datastore/src/index.js#L32-L33

Looks like it is a https://github.com/multiformats/multicodec/blob/master/table.csv#L174 //cc @kumavis

**Comment by kumavis** [comment]
hmm for the `state-trie` it should continue resolving as an `eth-account-snapshot`. A little confused how it ended up as a `eth-tx`.

**Comment by kumavis** [comment]
Sorry that i haven't been available to maintain the ipfs-eth stuff in a few weeks, hope to get back on it soon.

**Comment by kumavis** [comment]
Hmm seems i may have clobbered it here. the dangers of copy-pasting repo's and not updated the git remotes. Should extract all these eth-*-trie resolvers into a factory and then filling in the relevant types.

https://github.com/dignifiedquire/js-ipld-eth-state-trie/commit/bff107d62ca867850007dd518beb179c6b7c2d9c

File: src/index.js (hunk 2 above)
Comment (diasdavid) [inline src/index.js]: 👍

**Comment by diasdavid** [comment]
@kumavis can we get the remaining ipld-eth-x resolvers under ipld org? It makes it easier for all of us to help you maintain it.

Currently, it is blocking us from merging datastore all the way up.

**Comment by diasdavid** [comment]
Things missing for this PR:

- [ ] rm yarn.lock
- [ ] rebase master
- [ ] update ipld-eth-x packages //cc @kumavis

**Comment by diasdavid** [comment]
both me and @dignifiedquire (my handle is daviddias on npm http://npmjs.org/~daviddias)

**Comment by dignifiedquire** [comment]
@kumavis @diasdavid as mentioned in irc I'm still not sure how to fix the module, could you two please put your heads together and make it so? <3

**Comment by kumavis** [comment]
heres my status update:

- [x] js-ipld-eth-block
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [x] js-ipld-eth-tx
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [x] js-ipld-eth-state-trie
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [x] js-ipld-eth-storage-trie
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [x] js-ipld-eth-block-list
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [x] js-ipld-eth-account-snapshot
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [x] js-ipld-eth-trie
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [x] js-ipld-eth-tx-trie
  - [x] move to IPLD org
  - [x] add owners on npm
  - [x] sanity check + readme

- [ ] js-ipld-eth-tx-receipt-trie
  - [ ] implement

- [ ] js-ipld-eth-tx-receipt
  - [ ] implement

- [ ] General TODOs
  - [ ] new IpfsBlock constructor params
  - [ ] additionally resolve as field index (`<eth-block cid>/0`)

**Comment by diasdavid** [comment]
All right, thank you @kumavis! Now for the final trick, we just need to update all of these to the new block:

- [ ] https://github.com/ipld/js-ipld-eth-block
- [ ] https://github.com/ipld/js-ipld-eth-tx
- [ ] https://github.com/ipld/js-ipld-eth-trie
- [ ] https://github.com/ipld/js-ipld-eth-account-snapshot
- [ ] https://github.com/ipld/js-ipld-eth-state-trie
- [ ] https://github.com/ipld/js-ipld-eth-block-list
- [ ] https://github.com/ipld/js-ipld-eth-storage-trie
- [ ] https://github.com/ipld/js-ipld-eth-tx-trie

**Comment by dignifiedquire** [comment]
- https://github.com/ipld/js-ipld-eth-state-trie/pull/4
- https://github.com/ipld/js-ipld-eth-block/pull/17

**Comment by dignifiedquire** [comment]
🎉 🎉 🎉 🎉 🎉 🎉

**Pull request merged**